### ISSUE TITLE
Filtering out incoming events using middleware?

### ISSUE THREAD

**Issue by 15bubbles** [slackapi/bolt-python#484]
Hi,

me and my team are writing a Slack bot using your library and it helps us a lot, thumbs up for good work.

There's one thing I wanted to ask about and perhaps get some advice. We have a use case, where we want to skip some events from channels to which our bot was invited but these channels weren't configured in our registry.

I initially thought to filter out events that come for specific channel using global middleware and just not call `next` function when event came from not configured channel. Something more or less like this:

```
async def configured_channel_middleware(body, next, ack):
    if is_channel_configured(body):
        # only call `next` event came from configured channel
        await next()
    else:
        # artificially call `ack` so that event won't be retried
        await ack()

```

however using this we get such information in logs:
```
A global middleware (AsyncCustomMiddleware(func=ignore_not_configured_channel_events)) skipped calling either `next()` or `next_()` without providing a response for the request
```

Because of that I'm not sure if this is the best way to do this. Do you have any other recommendation how we could deal with such pattern?

**Comment by mwbrooks** [comment]
Hi @15bubbles it's great to hear that Bolt Python is helping you out - a huge shout out goes to @seratch for the amazing work on the framework!

Thanks for the detailed explaining of your app!

Have you had a chance to look at [Listener Matcher middleware](https://slack.dev/bolt-python/concepts#listener-middleware)? It allows you to define a matcher that can filter events. This may be work for your situation and it won't introduce a global middleware that may execute against non-event request. When the middleware returns false, it will not proceed.

**Comment by 15bubbles** [comment]
Hi @mwbrooks !

Thank you for reply!

Yes, I considered using listener matchers, however the reason why I didn't decide to use them is that they have to be declared on every listener on which they should be used and in our case there would be quite a few of them, so we decided to go with global middleware as we want to filter all requests (of course those that contain channel id in their payload but we're feeling safe for view submissions and options as these have to be triggered by some interactions which in turn have channel id in their payloads).

But I noticed that returning `BoltResponse` from middleware prevents those retries and we don't see warning messages anymore (I guess that I didn't put too much understanding to the error message itself as it mentions about providing response for the request).

So instead of doing this:
```
async def configured_channel_middleware(body, next, ack):
    if is_channel_configured(body):
        # only call `next` event came from configured channel
        await next()
    else:
        # artificially call `ack` so that event won't be retried
        await ack()
```

this works perfectly fine:
```
async def configured_channel_middleware(body, next, ack):
    if is_channel_configured(body):
        # only call `next` event came from configured channel
        return await next()
    else:
        # artificially call `ack` and return `BoltResponse` instance so the events won't be retried
        return await ack()
```

**Comment by seratch** [comment]
@15bubbles Yes, as you did, returning a response from a global middleware works for you. We're glad to hear that you've figured out what to do for your use case 🎉 

As it seems that everything is clear here, let us close this issue. But please feel free to write in whenever you have anything further regarding this topic.

**Comment by 15bubbles** [comment]
Yes, everything's clear

Thank you for your time @seratch @mwbrooks ! : D
